atecc608a_se.c

```diff
@@ -131,37 +148,29 @@ psa_status_t atecc608a_export_public_key(psa_key_slot_number_t key,
 {
     const size_t key_data_len = 65;
     const uint16_t slot = key;
-    ATCA_STATUS ret = ATCA_SUCCESS;
+    ATCA_STATUS ret = ATCA_GEN_FAIL;
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
 
     if (data_size < key_data_len)
     {
         return PSA_ERROR_BUFFER_TOO_SMALL;
     }
 
-    if (atcab_init(&atca_iface_config) != ATCA_SUCCESS)
-    {
-        return PSA_ERROR_HARDWARE_FAILURE;
-    }
+    ATCAB_INIT();
+
+    ASSERT_SUCCESS((ret = atcab_get_pubkey(slot, &p_data[1])), atecc608a_to_psa_error(ret));
```

Comment:
Suggestion: make a macro that embeds the `atecc608a_to_psa_error` transformation.